### PR TITLE
Fix-up coefficients and smallest denormal value for ieeeDouble in exp()

### PR DESCRIPTION
**Pull request by ibuclaw**
Addresses complaints in https://github.com/D-Programming-Language/phobos/pull/3285

Apart from the failing unittest....

### CODE REVIEW

**Comment by ibuclaw** [comment]
By the way, how do you guys get the represented hexadecimal float?  I've used ``"%.15a"`` format in the past, but it never prints a ``0x1....`` number for real types.  I do get plenty of ``0x9...`` and ``0xb...`` hexadecimal floats though. ;-)

File: std/math.d [hunk 1]
```diff
@@ -1574,36 +1574,55 @@ real exp(real x) @trusted pure nothrow @nogc
...
+            static immutable real[4] Q = [
+                2.0000000000000000000897E0L,
+                2.2726554820815502876593E-1L,
+                2.5244834034968410419224E-3L,
+                3.0019850513866445504159E-6L,
+            ];
+
+            // C1 + C2 = LN2.
+            enum real C1 = 6.9314575195312500000000E-1L;
+            enum real C2 = 1.4286068203094172321215E-6L;
```
Comment (ibuclaw) [inline std/math.d]: FYI, ``C2`` was changed from the (128bit number) ``1.428606820309417232121458176568075500134E-6L;``, if we are going to have a branch for each float format, might as well have the number in the correct precision.

**Comment by ibuclaw** [comment]
Ping.  @MartinNowak - this is for 2.068 release

**Comment by ibuclaw** [comment]
Also need to sort out https://issues.dlang.org/show_bug.cgi?id=14732

**Comment by ibuclaw** [comment]
Ping?

**Comment by DmitryOlshansky** [comment]
I'd opportunistically merge but have literally no iidea if this pull does the right thing. You might want to target the stable branch for 2.068 though.

File: std/math.d [hunk 2]
```diff
@@ -1574,36 +1574,55 @@ real exp(real x) @trusted pure nothrow @nogc
...
+            static immutable real[3] P = [
+                9.9999999999999999991025E-1L,
+                3.0299440770744196129956E-2L,
+                1.2617719307481059087798E-4L,
+            ];
+            static immutable real[4] Q = [
+                2.0000000000000000000897E0L,
+                2.2726554820815502876593E-1L,
+                2.5244834034968410419224E-3L,
+                3.0019850513866445504159E-6L,
```
Comment (MartinNowak) [inline std/math.d]: Those are the same coefficients as above.

File: std/math.d [hunk 3]
```diff
@@ -1574,36 +1574,55 @@ real exp(real x) @trusted pure nothrow @nogc
...
+            ];
+
+            // C1 + C2 = LN2.
+            enum real C1 = 6.9314575195312500000000E-1L;
+            enum real C2 = 1.4286068203094172321215E-6L;
+
+            // Overflow and Underflow limits.
+            enum real OF =  1.1356523406294143949492E4L;  // ln((1-2^-64) * 2^16384)
+            enum real UF = -1.13994985314888605586758E4L; // ln(2^-16446)
```
Comment (MartinNowak) [inline std/math.d]: Why not using hex float to avoid any parsing inaccuracies?

**Comment by MartinNowak** [comment]
Please rebase and repopen this against stable.

File: std/math.d (hunk 2 above)
Comment (ibuclaw) [inline std/math.d]: Yes, the coefficients should be identical regardless of what platform / float format we are using.

File: std/math.d (hunk 3 above)
Comment (ibuclaw) [inline std/math.d]: Could be worth address this in a separate PR to cover all of them?
